### ISSUE TITLE
Initializing a non-entity ref type in select creates a single instance for every record

### ISSUE THREAD

**Issue by kszicsillag** [dotnet/efcore#12274]
Consider the following entity type and its DTO type.
```csharp
// Entity
public class Dog
{
    public int Id { get; set; }
    public string Name { get; set; }
}

//DTO
 public class DogDTO
 {
     public int ID { get; set; }
     public string  Name { get; set; }
     public IDCard IDCard { get; set; }
 }

 public class IDCard
 {
     public int ID { get; set; }
     public string Number { get; set; }
 }
```
Now query all the entities and project them to the DTO type.

```csharp
 var dogDTOList = ctx.Dogs.Select(
   d => 
     new DogDTO { ID = d.Id
    ,Name = d.Name
    ,IDCard = new IDCard() })
.ToList();
```
We would expect that a new IDCard instance will be created for each DogDTO - as this is what would happen in case of LINQ-to-Objects.
However, the constructor of IDCard is called only once and the IDCard instance will be the same for all DogDTOs.

There is a full repro repo [here](https://github.com/kszicsillag/efcore-selectinit).

### Further technical details
EF Core version: 2.1.0
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows Server 2016
IDE: Visual Studio 2017 15.7

**Comment by kszicsillag** [comment]
We observed that if the constructor is called with different parameter values for every record (e.g. with some data from the record) we get the expected beahvior: different instance created for every record.

**Comment by smitpatel** [comment]
Duplicate of #7983

**Comment by smitpatel** [comment]
`IDCard` does not depend on the range variable hence being evaluated beforehand causing it to have same object for each result. There is no easy way to figure out this case.

**Comment by kszicsillag** [comment]
Thanks for pointing me to https://github.com/aspnet/EntityFrameworkCore/issues/7983.
If this won't be  fixed, at very least the documentation should warn about this issue, clearly state the limitations and give guidelines about how to write a projection without falling into this kind of pitfalls.

**Comment by ajcvickers** [comment]
Note from triage: we will reconsider this, perhaps pivoting on whether or no the call to new is in the final projection. Also, Relinq makes this problematic because it erases the New expression.

**Comment by smitpatel** [comment]
We have done work to now create parameter for NewExpression query tree. So this should likely be fixed.
